packages/lib/server-only/document/find-documents.ts

```diff
@@ -82,7 +92,33 @@ export const findDocuments = async ({
     })
     .otherwise(() => undefined);
 
-  const filters = team ? findTeamDocumentsFilter(status, team) : findDocumentsFilter(status, user);
+  const visibilityFilters = [
+    ...match(teamMemberRole)
+      .with(TeamMemberRole.ADMIN, () => [
+        { visibility: DocumentVisibility.EVERYONE },
+        { visibility: DocumentVisibility.MANAGERANDABOVE },
+        { visibility: DocumentVisibility.ADMIN },
+      ])
+      .with(TeamMemberRole.MANAGER, () => [
+        { visibility: DocumentVisibility.EVERYONE },
+        { visibility: DocumentVisibility.MANAGERANDABOVE },
+      ])
+      .otherwise(() => [{ visibility: DocumentVisibility.EVERYONE }]),
+    {
+      Recipient: {
+        some: {
+          email: user.email,
+        },
+      },
+    },
+  ];
+
+  const filters = team
```

Comment:
Should I also update the search function `searchDocumentsWithKeyword` with this logic?

At the moment, the function `searchDocumentsWithKeyword` returns all the documents **uploaded** by the current team member. See the image below.

![CleanShot 2024-08-05 at 10 59 06](https://github.com/user-attachments/assets/1a5241e7-83c5-47c6-b4c5-ca447e902a28)

In the above image, the 2 documents are the ones properly returned—the documents that can be accessed by the uploader, with the visibility set to `everyone`. However, the 4 documents returned by `searchDocumentsWithKeyword` are all the documents uploaded by the current team member, some of which are set to `admin` or `managerandabove`.

However, there might be edge cases where an admin or manager sets the visibility of a document uploaded by someone else to `admin` or `managerandabove`. In that case, the uploader shouldn't be able to access the document, right? Hence, the `searchDocumentsWithKeyword` function shouldn't return it.

I am asking here because I feel I'm overthinking a bit and want to clear it up.